ui/app/AppLayouts/Browser/BrowserLayout.qml

```diff
@@ -122,16 +122,656 @@ Item {
-            // In the meantime, I'm opening the content in the same webengineview
-            request.openIn(browserContainer)
+            var tab = tabs.createEmptyTab(currentWebView.profile);
+            tabs.currentIndex = tabs.count - 1;
+            request.openIn(tab.item);
+        }
+    }
+
+    MessageDialog {
+        id: sslDialog
+
+        property var certErrors: []
+        icon: StandardIcon.Warning
+        standardButtons: StandardButton.No | StandardButton.Yes
+        title: qsTr("Server's certificate not trusted")
+        text: qsTr("Do you wish to continue?")
+        detailedText: qsTr("If you wish so, you may continue with an unverified certificate. Accepting an unverified certificate means you may not be connected with the host you tried to connect to.\nDo you wish to override the security check and continue?")
+        onYes: {
+            certErrors.shift().ignoreCertificateError();
+            presentError();
+        }
+        onNo: reject()
+        onRejected: reject()
+
+        function reject(){
+            certErrors.shift().rejectCertificate();
+            presentError();
+        }
+        function enqueue(error){
+            certErrors.push(error);
+            presentError();
+        }
+        function presentError(){
+            visible = certErrors.length > 0
+        }
+    }
+
+    DownloadView {
+        id: downloadView
+        visible: false
+        anchors.fill: parent
+    }
+
+    function onDownloadRequested(download) {
+        console.log("DOWNLOAD REQUESTED")
+        downloadView.visible = true;
+        downloadView.append(download);
+        download.accept();
+    }
+
+    FindBar {
+        id: findBar
+        visible: false
+        anchors.right: parent.right
+        anchors.rightMargin: 10
+        anchors.top: parent.top
+
+        onFindNext: {
+            if (text)
+                currentWebView && currentWebView.findText(text);
+            else if (!visible)
+                visible = true;
+        }
+        onFindPrevious: {
+            if (text)
+                currentWebView && currentWebView.findText(text, WebEngineView.FindBackward);
+            else if (!visible)
+                visible = true;
+        }
+    }
+
+    Rectangle {
+        id: statusBubble
+        color: "oldlace"
+        property int padding: 8
+        visible: false
+
+        anchors.left: parent.left
+        anchors.bottom: parent.bottom
+        width: statusText.paintedWidth + padding
+        height: statusText.paintedHeight + padding
+
+        Text {
+            id: statusText
+            anchors.centerIn: statusBubble
+            elide: Qt.ElideMiddle
+
+            Timer {
+                id: hideStatusText
+                interval: 750
+                onTriggered: {
+                    statusText.text = "";
+                    statusBubble.visible = false;
+                }
+            }
         }
```

Comment:
This file is massive. We should probably split it at some point is multiple subcomponents even if they are not reused somewhere else